Update dockerfiles to use go v1.22

aggarwal0009 opened the pull request:
<!-- Thank you for helping Azure Container Networking with a pull request!
Use conventional commit messages, such as
  feat: add a knob to the frobnitz
or
  fix: repair hole in wumpus
And read this for faster PR reviews: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#best-practices-for-faster-reviews -->

**Reason for Change**:
Upgrading dockerfiles to go v1.22 to support device plugin which introduces kubelet dependencies requiring go 1.22

Reference to device plugin PR: https://github.com/Azure/azure-container-networking/pull/2886/

**Issue Fixed**:
<!-- If this PR fixes GitHub issue 1234, add "Fixes #1234" to the next line. -->


**Requirements**:
<!-- Put an "X" character inside the brackets of each completed task. Some may be optional depending on the PR. -->


- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  <!-- Common commit types:
        build: Build 🏭
        chore: Maintenance 🔧
        ci: Continuous Integration 💜
        docs: Documentation 📘
        feat: Features 🌈
        fix: Bug Fixes 🐞
        perf: Performance Improvements 🚀
        refactor: Code Refactoring 💎
        revert: Revert Change ◀️
        style: Code Style 🎶
        security: Security Fix 🛡️
        test: Testing 💚 -->
- [x] includes documentation
- [ ] adds unit tests
- [] relevant PR labels added

**Notes**:

matmerr commented:
1.23 is out, any reason not going all the way up?

https://go.dev/blog/go1.23

rbtr commented:
> 1.23 is out, any reason not going all the way up?
> 
> https://go.dev/blog/go1.23

we should probably wait for at least 1.23.1 in case they snuck any bugs in the .0, but this is a good idea. some nice new features in 1.23.

@aggarwal0009 8 files does not seem like enough [compared to when we moved to 1.21](https://github.com/Azure/azure-container-networking/pull/2384/files#diff-421301c67ad8e21cdff91d05da96951b255ac2c63f6fdf337283010fd78b39f3)

aggarwal0009 commented:
> > 1.23 is out, any reason not going all the way up?
> > https://go.dev/blog/go1.23
> 
> we should probably wait for at least 1.23.1 in case they snuck any bugs in the .0, but this is a good idea. some nice new features in 1.23.
> 
> @aggarwal0009 8 files does not seem like enough [compared to when we moved to 1.21](https://github.com/Azure/azure-container-networking/pull/2384/files#diff-421301c67ad8e21cdff91d05da96951b255ac2c63f6fdf337283010fd78b39f3)

Updated

aggarwal0009 commented:
/azp run Azure Container Networking PR

rbtr reviewed (approved):
I'll update the PR check requirements once this tests clean so it can be queued to merge

aggarwal0009 commented:
/azp run Azure Container Networking PR

aggarwal0009 commented:
/azp run Azure Container Networking PR

timraymond commented:
#2930 depends on this (adding it to link the PRs)

aggarwal0009 commented:
/azp run Azure Container Networking PR

aggarwal0009 commented:
/azp run Azure Container Networking PR

aggarwal0009 commented:
/azp run Azure Container Networking PR

rbtr commented on Makefile:
```diff
@@ -874,7 +874,7 @@ CNI_TYPE ?= cilium
 # COVER_FILTER omits folders with all files tagged with one of 'unit', '!ignore_uncovered', or '!ignore_autogenerated'
 test-all: ## run all unit tests.
-	@$(eval COVER_FILTER=`go list --tags ignore_uncovered,ignore_autogenerated $(COVER_PKG)/... | tr '\n' ','`)
+	@$(eval COVER_FILTER=`go list $(COVER_PKG)/... | tr '\n' ','`)
```
everything else looks fine, but what is the effect of this change?

aggarwal0009 replied:
I think this will affect the code coverage % somewhat but this was causing one of the flaky tests to be fail repeatedly 
https://teams.microsoft.com/l/message/19:2b104ee19fdb450b807903eb7c79588a@thread.skype/1724175877025?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=1eb27b57-8ace-4f0e-a9a9-e6db4e9829b7&parentMessageId=1724175877025&teamName=ContainerNetworking&channelName=CNI&createdTime=1724175877025

aggarwal0009 replied:
@rbtr  I checked the coverage % diff and its negligible. 
Without makefile change: https://msazure.visualstudio.com/One/_build/results?buildId=101038052&view=logs&j=792982d5-3bb1-5d82-222e-228148b73448&t=fe31bb4d-d8c6-5505-3ab3-3c9a2560c42b

With Makefile change:  https://msazure.visualstudio.com/One/_build/results?buildId=101344151&view=logs&j=792982d5-3bb1-5d82-222e-228148b73448&t=fe31bb4d-d8c6-5505-3ab3-3c9a2560c42b

QxBytes replied:
can we update the comment too to reflect what we're including in testing/coverage

aggarwal0009 replied:
updated

aggarwal0009 commented:
/azp run Azure Container Networking PR

aggarwal0009 commented:
/azp run Azure Container Networking PR

huntergregory reviewed (changes requested):
Since this is modifying NPM, please run both NPM Azure Pipelines tests and merge once green. They are not “required checks” because GitHub does not support requiring checks that trigger conditionally (i.e. only when npm/ directory is modified)

/azp run NPM Conformance Tests
/azp run NPM Scale Test

huntergregory commented:
/azp run NPM Conformance Tests

huntergregory commented:
/azp run NPM Scale Test
